webpack/webpack-tests.ts

```diff
@@ -480,3 +482,7 @@ configuration = {
 		]
 	}
 }
+
+const resolve: webpack.Resolve = {
+    cachePredicate: 'boo' // why does this test _not_ fail!?
```

Comment:
Why does this test not fail? The type of `webpack.Resolve` is `webpack.NewResolve | webpack.OldResolve`; only `webpack.NewResolve` has a potential `cachePredicate` key, so by narrowing the RHS should be treated as a declaration of `webpack.NewResolve`, but it's being treated as `any`.